### DISCUSSION TITLE
Is searching by description planned to come back?

### DISCUSSION THREAD

**Discussion by BlueWolfProduction** [immich-app/immich#9143] [Q&A] [answered]
I discovered Immich relative recently and loved it ever since. I uploaded a total of 13076 pictures on it and probably tagged at least half of them with things I care about in the description (mainly dog names, which there are a lot of). That way I can easily search by dogs which I quite often have to do for my work. Probably took me 2-3 weeks to complete, it was an ungodly amount of work, but the end result was totally worth it by giving me a smooth experience to navigate through my large library.

...And then a week after I completed that there was an update that revamped the entire search, and removed the ability to search by description. Just my luck of course! I know this is all a work in progress and search features are slowly coming back. I have been waiting patiently ever since for it to come back, but I haven't seen any signs of searching by description coming back. Many times I could have really used that feature now, but I was unable to search through my pictures. I have so many tagged photos I cannot locate currently. I don't think there is any other way for me to tag photos other than through the description?

Long story short: I am just wondering if searching by description is indeed planned to come back. And if there is an ETA for it at all.

**Comment by mertalev** [comment, marked as answer]
We plan to support this at some point, but haven't decided on the direction we want to go with it yet. It could be in the form of a "full-text search" that combines filename search, image description, etc. in the same search bar, it could be in the form of tags/chips that let users search a specific field, possibly both. There are tradeoffs to each that we'd want to consider first. I don't think it'll happen in the short term unless someone volunteers to work on it, but maybe in a few months?

**Reply by BlueWolfProduction** [reply]
Thanks for your answer!

**Reply by elephantchau** [reply]
Search by descriptions is such a hugely wanted basic function. It'll be amazing to be able to match by any field if possible.

**Comment by mmomjian** [comment]
For now I added a section to the DB queries in case that helps you: https://immich.app/docs/guides/database-queries
